docs/src/appendix/forge-library/signature/secp256_curve.md

```diff
@@ -0,0 +1,82 @@
+# `Secp256 Curve`
+
+This module contains the implementation of `KeyPairTrait` for the `Secp256k1` and `Secp256r1` curves.
+
+> ⚠️ **Security Warning**
+>
+> Please note that cryptography in Starknet Foundry is still experimental and **has not been audited yet**.
+>
+> Use at your own risk!
+
+
+## `Secp256CurveKeyPairImpl`
+
+The `Secp256CurveKeyPairImpl` is an implementation of `KeyPair` with secret key being `u256` and private key being a `Secp256Point`.
+
+The `Secp256Point` implements `Secp256PointTrait` from corelib.
+
+```rust
+struct KeyPair {
+    secret_key: u256,
+    public_key: Secp256Point,
+}
+```
+
+A new key pair for the `Secp256r1` curve can be obtained as follows:
+
+```rust
+use snforge_std::signature::{KeyPair, KeyPairTrait, SignerTrait, VerifierTrait};
+use snforge_std::signature::secp256_curve::{Secp256CurveKeyPairImpl, Secp256CurveSignerImpl, Secp256CurveVerifierImpl};
+use starknet::secp256r1::{Secp256r1Impl, Secp256r1Point, Secp256r1PointImpl};
+
+let key_pair_secp256r1 = KeyPairTrait::<u256, Secp256r1Point>::generate();
+```
+
+Or, if the `secret_key` is known in advance:
+
+```rust
+use snforge_std::signature::{KeyPair, KeyPairTrait, SignerTrait, VerifierTrait};
+use snforge_std::signature::secp256_curve::{Secp256CurveKeyPairImpl, Secp256CurveSignerImpl, Secp256CurveVerifierImpl};
+use starknet::secp256r1::{Secp256r1Impl, Secp256r1Point, Secp256r1PointImpl};
+
+let secret_key = ...;
+let key_pair_secp256r1 = KeyPairTrait::<u256, Secp256r1Point>::from_secret_key(secret_key);
+```
```

Comment:
Please drop this part, it is unnecessary wordy, global example + signatures are enough.